src/ranking.jl

```diff
@@ -171,3 +171,174 @@ Missing values are assigned rank `missing`.
 """
 tiedrank(x::AbstractArray; sortkwargs...) =
     _rank(_tiedrank!, x, Float64; sortkwargs...)
+
+"""
+    quantilerank(v, value; method=:inc)
+
+Compute the quantile(s)-position in [0, 1] interval of a `value` relative to a collection 
+`v`, e.g. a quantile rank of x means that (x*100)% of the elements in `v` are lesser 
+(strict) or lesser-equal (weak) the given `value`. 
```

Comment:
I haven't found a more general definition than this and each method gives a specific definition, according to the method.

Since this definition it's general, maybe we could omit the definition in parentheses and write something like this

"Compute the quantile(s)-position in [0, 1] interval of a `value` relative to a collection `itr`. Generally speaking, we can interpret the result of a quantile ranking of x to mean that (x*100 )% of the elements in `itr` are less than or equal to the given `value`, according to the respective method used."

Reply:
Does this definition really applies to all methods? AFAICT it's not the case unfortunately.

Reply:
Looking all the options in returns (except interpolations), we have:

```
return count_less / (n - 1)                          # :inc
return (count_less + 1) / (n + 1)                # :exc
value ∈ itr && (count_less += 1)              # :compete
return (count_less - 1) / (n - 1)
return (count_less + count_equal/2) / n   # :tied
return count_less / n                                 # :strict
return (count_less + count_equal) / n      # :weak
```

So, since the biggest differences between the methods are in the numerators and in all the denominators we use the variables `count_less` and `count_equal`, I think it seems reasonable to say that X% of the elements are less than or less-equal than "value" (according to each method).

What do you think? If you prefer, we could discard this sentence.

Reply:
OK, I guess that's reasonable. Though what's more correct, saying "less than or equal" or just "less than"? The former seems more precise, but that precision could be misleading since it doesn't work for `:strict`?

Reply:
I also prefer the first option. 

Thinking about what you said about `:strict`, to me, it seems right to me something like this.

"..., e.g. a quantile rank of x means that (x*100)% of the elements in `itr` **are less than 
or less-equal than** the given `value`, according to the respective method used."

This way we encapsulate the cases which is < (strict) and <= (all other methods).

What do you think?

Reply:
Or maybe "are less than (or less than or equal)"?

Reply:
I like that. I'll include in the next commit.

Reply:
Done in commit b74e8dc.